module/zfs/zthr.c

```diff
@@ -207,6 +207,9 @@ struct zthr {
 	/* flag set to true if we are canceling the zthr */
 	boolean_t	zthr_cancel;
 
+	/* flag set to true if we are waiting for the zthr to finish */
+	boolean_t	zthr_finish;
+	kcondvar_t	zthr_finish_cv;
```

Comment:
I'm trying to remember why we decided to not name this some variation of `zthr_wait`.
If it is only the fact that we don't want to associate it with `zthr_wait_time` which is something different, then it may worth to change that variable name, and keep `zthr_wait` for our variables.

Maybe we can we can do the following changes:
`zthr_wait_time` -> `zthr_sleep_timeout`

`zthr_finish_cv` -> `zthr_wait_cv`/`zthr_waiter_cv`
`zthr_finish` -> `zthr_haswaiters`
`zthr_finish()` -> `zthr_wait_cycle_done()`
`zthr_isfinishing()` -> `zthr_has_waiters()`

The proposed names are not absolute, I'm just afraid that with the current code `finish` may be confused with `cancel`. Let me know what you think.

Reply:
Yeah, I think that's a reasonable proposal. I think mostly I was dissatisfied with the ambiguity of the `zthr_wait*` names I had been using and want to go for something a little more specific. I think your suggestion addresses that and more accurately reflects that this is "waiting-type" behavior, not cancelling.